### PR TITLE
Fix API reference generation (quartodoc)

### PR DESCRIPTION
**Pull request by cpsievert**
This PR fixes `make docs`

<details>
<summary>Thanks to some help from @machow:</summary>

Hmmm... debugging with

```
from quartodoc import get_object

get_object("shiny:ui.page_sidebar", dynamic=True)
```

When I go up and print out the griffe object trying to get ui, I see this:

```
ipdb> !self
Module([PosixPath('/Users/machow/repos/py-shiny/venv/bin/shiny')])
```

But I feel like that's not the right path for shiny. I wonder if there's some kind of venv thing that's putting this path too early or something... here's my sys.path

```
['/Users/machow/repos/py-shiny/venv/bin', '/Users/machow/.pyenv/versions/3.12.3/lib/python312.zip', '/Users/machow/.pyenv/versions/3.12.3/lib/python3.12', '/Users/machow/.pyenv/versions/3.12.3/lib/python3.12/lib-dynload', '', '/Users/machow/repos/py-shiny/venv/lib/python3.12/site-packages', '__editable__.shiny-0.9.0.9000.finder.__path_hook__']
```

idk if bin was always there (and it makes sense it would be early in the path), but I suspect if it weren't in the path then it would find what it needed in site-packages

It's also possible there's a pip issue going on, since there have been issues with editable installs in pip in the past, so I wonder if the passing one uses an earlier version of pip, if locking to that would fix

</details>

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -114,8 +114,8 @@ doc =
     tabulate
     shinylive
     pydantic==1.10
-    quartodoc==0.7.2
-    griffe==0.33.0
+    quartodoc
+    griffe
```
Comment (cpsievert) [inline setup.cfg:118]:
```suggestion
    quartodoc==0.7.2
    griffe==0.33.0
```

File: docs/Makefile [hunk 2]
```diff
@@ -46,8 +46,8 @@ dev-shinylive: $(PYBIN)  ## Install development version of shinylive
 	$(PYBIN)/pip install https://github.com/posit-dev/py-shinylive/tarball/main
 
 deps: $(PYBIN) dev-htmltools dev-shinylive ## Install build dependencies
-	$(PYBIN)/pip install pip --upgrade
-	$(PYBIN)/pip install -e ..[doc]
+	$(PYBIN)/pip install --force-reinstall -v "pip<24.0"
```
Comment (cpsievert) [inline docs/Makefile:49]:
```suggestion
	$(PYBIN)/pip install pip --upgrade
```

**Comment by wch** [comment]
Strange, but 👍 .

**Review by wch — APPROVED** [review]

**Comment by gadenbuie** [comment]
Great debugging! I think we need this over in posit-dev/py-shiny-site too

**Pull request merged**